js_modules/dagster-ui/packages/ui-core/src/app/AppProvider.tsx

```diff
@@ -109,14 +111,19 @@ export const AppProvider = (props: AppProviderProps) => {
   const retryLink = React.useMemo(() => {
     return new RetryLink({
       attempts: {
-        max: 2,
-        retryIf: (error, _operation) => {
-          return error && error.statusCode && [502, 503, 504].includes(error.statusCode);
+        max: 3,
+        retryIf: async (error, _operation) => {
+          if (error && error.statusCode && httpStatusCodesToRetry.includes(error.statusCode)) {
+            // Retry-after header is in seconds, convert to ms by multiplying by 1000.
+            const wait = parseFloat(error?.response?.headers?.get?.('retry-after') ?? '0.3') * 1000;
+            await new Promise((res) => {
+              setTimeout(res, wait);
+            });
```

Comment:
pausing in `retryIf` feels like a weird way to do this

should we be using a `delay` function instead https://www.apollographql.com/docs/react/api/link/apollo-link-retry/#custom-strategies ?

Reply:
I did try that but it looks like it was ignoring it, I think these options might be mutually exclusive but didn't dig into it.

Reply:
`retryIf` is explicitly typed to allow returning a Promise so I think the idea is to combine `attempt` and `delay` into a single option

Reply:
> I did try that but it looks like it was ignoring it

was this before we were converting seconds to ms correctly? i could see that make it seem like it was ignored 

> so I think the idea is to combine attempt and delay into a single option

I'm not sure i follow since `delay` and `attempts` are separate sections and `retryIf` is just part of the `attempts` section

Reply:
looking at the source it seems pretty clear that the delay function is the intended tool 

https://github.com/apollographql/apollo-link/blob/master/packages/apollo-link-retry/src/retryLink.ts
https://github.com/apollographql/apollo-link/blob/master/packages/apollo-link-retry/src/delayFunction.ts 

 in this current form I believe you are going to get this `retryIf` sleep combined with the default delay function

Reply:
I will add a debugger to verify but from my understanding if we want to conditionally retry we need to use `retryIf` which from my testing is mutually exclusive with the other options.